pkg/asset/installconfig/clusterid.go

```diff
@@ -1,30 +1,70 @@
 package installconfig
 
 import (
+	"fmt"
+	"regexp"
+
 	"github.com/pborman/uuid"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
 
 	"github.com/openshift/installer/pkg/asset"
 )
 
+const (
+	// resource using HumanID usually have suffixes like `[-/_][a-z]{3,4}` eg. `_int`, `-ext` or `-ctlp`
+	maxNameLen = 32 - 5
+	randomLen  = 5
+	maxBaseLen = maxNameLen - randomLen - 1
+)
+
 // ClusterID is the unique ID of the cluster, immutable during the cluster's life
 type ClusterID struct {
-	ClusterID string
+	// UUID is a globally unique identifier.
+	UUID string
+
+	// HumanID is an identifier for the cluster that is more human friendly.
+	// This does not have
+	HumanID string
 }
 
 var _ asset.Asset = (*ClusterID)(nil)
 
-// Dependencies returns no dependencies.
+// Dependencies returns install-config.
 func (a *ClusterID) Dependencies() []asset.Asset {
-	return []asset.Asset{}
+	return []asset.Asset{
+		&InstallConfig{},
+	}
 }
 
-// Generate generates a new UUID
-func (a *ClusterID) Generate(asset.Parents) error {
-	a.ClusterID = uuid.New()
+// Generate generates a new ClusterID
+func (a *ClusterID) Generate(dep asset.Parents) error {
+	ica := &InstallConfig{}
+	dep.Get(ica)
+
+	// add random chars to the end to randomize
+	a.HumanID = generateHumanID(ica.Config.ObjectMeta.Name)
+	a.UUID = uuid.New()
 	return nil
 }
 
 // Name returns the human-friendly name of the asset.
 func (a *ClusterID) Name() string {
 	return "Cluster ID"
 }
+
+// generateHumanID take base and returns a ID that
+// - is of max length maxBaseLen + randomLen + 1
```

Comment:
`is of max length maxNameLen`

Reply:
Oh Yeah :) will fix that.

Reply:
done with https://github.com/openshift/installer/pull/1280/commits/9f5c0c00348230fdf601bb466b2793638e852a8b